src/infractl/deploy/gcp/main.py

```diff
@@ -0,0 +1,118 @@
+            "nvidia-tesla-v100",
+            "nvidia-tesla-p100",
+            "nvidia-tesla-k80",
+        ],
+        "n1-highcpu": [
+            "nvidia-tesla-t4",
+            "nvidia-tesla-p4",
+            "nvidia-tesla-v100",
+            "nvidia-tesla-p100",
+            "nvidia-tesla-k80",
+        ],
+    }
+    zone = os.environ["ICL_GCP_ZONE"]
+    machine_type = os.environ["ICL_GCP_MACHINE_TYPE"]
+
+    # Check machine type availability
+    if machine_type:
+        command = [
+            "gcloud",
+            "compute",
+            "machine-types",
+            "list",
+            "--filter",
+            f"name='{machine_type}'",
+            "--format=value(name)",
+            f"--zones={zone}",
+        ]
+        try:
+            result = run(command, check=True, capture_output=True, text=True)
+            if machine_type not in result.stdout:
+                print(f"Error: {machine_type} is not available in {zone}.")
+                print("Please try a different machine type or zone.")
+                sys.exit(100)
+            else:
+                print("Machine is available")
+        except CalledProcessError:
+            # If the command failed, print an error message
+            print("Failed to check machine type availability.")
+            sys.exit(10)
+
+        # Check GPU model availability
+        command = ["gcloud", "compute", "accelerator-types", "list", "--verbosity=error"]
+        try:
+            result = run(command, check=True, capture_output=True, text=True)
+            if gpu_model not in result.stdout:
+                print(f"Error: {gpu_model} is not available in {zone}.")
+                print("Please try a different gpu model or zone.")
+                sys.exit(200)
+            else:
+                print("GPU is available")
+        except CalledProcessError:
+            # If the command failed, print an error message
+            print("Failed to check gpu model availability.")
+            sys.exit(11)
+    else:
+        print("The 'machine_type' variable is empty")
+
+    # Check machine type and gpu model compatibility
+    # Simplify the machine type to match keys in supported_gpus_dict
+    # For example, convert "n1-standard-8" to "n1-standard"
+    simplified_machine_type = machine_type.rsplit("-", 1)[0]
+
+    # Check if simplified_machine_type is a key in supported_gpus_dict
+    if simplified_machine_type in supported_gpus_dict:
+        # Check if gpu_model is in the list associated with the key
+        if gpu_model in supported_gpus_dict[simplified_machine_type]:
+            print(f"Machine type {machine_type} supports the {gpu_model} GPU.")
+        else:
+            print(f"Error: Machine type {machine_type} doesn't support the {gpu_model} GPU.")
+            sys.exit(100)
+    else:
+        print(
+            f"Error: Machine type \"{machine_type}\" \
+            (simplified to {simplified_machine_type}) is not recognized."
```

Comment:
See https://google.github.io/styleguide/pyguide.html#s3.2-line-length
> Do not use a backslash for explicit line continuation
```suggestion
            f"Error: Machine type \"{machine_type}\"
            f"(simplified to {simplified_machine_type}) is not recognized."
```

Reply:
Resolved in commit:
https://github.com/intel-ai/icl/pull/76/commits/0de90d5ba55ef61cead0e0144be856e661c7e49d